src/tools/cephfs/top/cephfs-top

```diff
@@ -413,7 +413,9 @@ class FSTop(FSTopBase):
                 endmenu = True
             elif key == ord('q'):
                 self.stdscr.erase()
-                if self.active_screen == FS_TOP_ALL_FS_APP:
+                if not fs_list:  # no fs, navigate to home screen
+                    self.run_all_display()
+                elif self.active_screen == FS_TOP_ALL_FS_APP:
```

Comment:
It looks better. Can't we have the above `if` and `elif` in a single check with `or` as they are supposed to perform same action.

Reply:
No, it should be separate. We need to check `self.active_screen == FS_TOP_ALL_FS_APP` only if `fs_list` is True.

Reply:
oops!! sorry...I missed that case

Reply:
Did better fix